### PR TITLE
Alerting: Support tls config for webhook receiver

### LINKED ISSUE #9548: Support self signed certs for alert notifications
- What Grafana version are you using? `grafana/grafana:master`
- What datasource are you using? `Prometheus`
- What OS are you running grafana on? `CentOS 7 on virtualBox` 
- What did you do? `Tried to add a notification alert channel via the grafana UI`
- What was the expected result? `A notification would be sent`
- What happened instead? 

```
grafana_1        | t=2017-10-16T18:06:24+0000 lvl=eror msg="Failed to send alert notifications" logger=context userId=1 orgId=1 uname=admin error="Post https://mattermost.XXXXXXXXXX.com/hooks/<hook id>: x509: certificate signed by unknown authority"
grafana_1        | t=2017-10-16T18:06:24+0000 lvl=eror msg="Request Completed" logger=context userId=1 orgId=1 uname=admin method=POST path=/api/alert-notifications/test status=500 remote_addr=X.X.X.X time_ms=26 size=48 referer=http://localhost:3000/alerting/notification/1/edit
 ```

To troubleshoot we manually added our certificates with Grafana running in a Docker container:

 `docker run --rm --entrypoint="/bin/bash" -p 3000:3000 -it grafana/grafana`

Our signed certificates were added to `/etc/ssl/certs` and then we ran `c_rehash /etc/ssl/certs`

We curled our mattermost instance `curl https://mattermost.ourdomain.com` successfully:

```
root@4b3ababf117c:/# curl https://mattermost.ourdomain.com
<!DOCTYPE html> <html> <head> <meta http-equiv=X-UA-Compatible content="IE=edge"> <meta name=viewport content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=0"> ....
```

Is there some way to use our self signed certificates with Grafana to add an alerting channel?

### PR DESCRIPTION
**Pull request by titolins**
Takes advantage of https://github.com/grafana/grafana/pull/92331, applying basically the same logic from mqtt receivers into webhook to fix #9548.

## Related PR's
- https://github.com/grafana/alerting/pull/247
- https://github.com/grafana/mimir/pull/9341

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -10,6 +10,8 @@ replace cuelang.org/go => github.com/grafana/cue v0.0.0-20230926092038-971951014
 // import that instead of v0.X even though v0.X is newer.
 replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.52.0
 
+replace github.com/grafana/alerting => /Users/titolins/dev/alerting
```
Comment (titolins) [inline go.mod:13]: Leaving this here just as a reminder to update alerting once https://github.com/grafana/alerting/pull/247 has been merged

**Review by fayzal-g — APPROVED** [review]
LGTM, just a couple nits

**Review by fayzal-g — CHANGES_REQUESTED** [review]
Removing previous approval, as it needs the PR from grafana/alerting to go in first. Will re-review once that's done

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -200,6 +200,7 @@
 /devenv/docker/blocks/memcached/ @grafana/grafana-backend-group
 /devenv/docker/blocks/mimir_backend/ @grafana/alerting-backend
 /devenv/docker/blocks/mqtt/ @grafana/alerting-backend
+/devenv/docker/blocks/caddy_tls/ @grafana/alerting-backend
```
[the remark this answers is not in the document]
Comment (titolins) [reply .github/CODEOWNERS]: Ah, my bad - hadn't noticed that. Fixed now 👍

**Comment by brendamuir** [comment]
docs lgtm

**Review by fayzal-g — APPROVED** [review]
Small typo in the doc but lgtm!

**Pull request merged**